lib/PreferenceList.cc

```diff
@@ -47,6 +48,40 @@ PrefListElement PreferenceList::at(SizeType index) const {
     return pref_list_.at(index);
 }
 
+void PreferenceList::set_ties(RankType rank, PreferenceList p){
+    SizeType size = p.size();
+    cur_rank_++;
+    is_tied_[rank] = true;
+    for(SizeType i=0; i<size; i++){
+        tie_map_[rank].emplace_back(rank, p.at(i).vertex); //the rank for the preference list element is index 
+    }
+}
+
+PreferenceList::ContainerType PreferenceList::get_ties(RankType rank) const {
+    return tie_map_.at(rank);
+}
+
+bool PreferenceList::isTied(RankType rank) const {
+    return is_tied_.at(rank);
+}
+
+void PreferenceList::printList(){
```

Comment:
This function should be merged with operator<<. It should not print to standard output.